### PR TITLE
Population update

### PR DESCRIPTION
**Pull request by isunix11**

### CODE REVIEW

File: data/population.json [hunk 1]
```diff
@@ -1048,5 +1048,5 @@
       "CountryName": "Zimbabwe",
       "Population": 14439018
     }
-  }
+
```
Comment (thepassle) [inline data/population.json:1051]: There are quite some keys in this file that we don't use which is a bit wasteful, could we optimize this somehow?
e.g.:
```json
    ":::Early-demographic dividend": {
      "CountryName": "Early-demographic dividend",
      "Population": 3249140605
    },
    ":::East Asia & Pacific": {
      "CountryName": "East Asia & Pacific",
      "Population": 2328220870
    },
    ":::East Asia & Pacific (excluding high income)": {
      "CountryName": "East Asia & Pacific (excluding high income)",
      "Population": 2081651801
    },
    ":::East Asia & Pacific (IDA & IBRD countries)": {
      "CountryName": "East Asia & Pacific (IDA & IBRD countries)",
      "Population": 2056064424
    },
```

File: src/components/CountryInfo.js [hunk 2]
```diff
@@ -129,7 +129,7 @@ export class CountryInfo extends Component {
           <h2>Stats</h2>
```
Comment (thepassle) [inline src/components/CountryInfo.js:129]: There's an unused check now: https://github.com/Code-for-All/lockdown/blob/7f8e86f7e6f383c1ad9d75dea8449d692db4b937/src/components/CountryInfo.js#L96

Have you tested this when the user doesnt have internet connection, too?

File: src/services/populationService.js [hunk 3]
```diff
@@ -1,39 +1,15 @@
...
-
-        return this.cache[iso2];
-      } catch (_) {
-        this.cache[iso2] = {
-          status: 'failed'
-        };
-      }
+  async getPopulation(forceRefresh) {
+    if (forceRefresh || !this.__population) {
+      this.__population = fetch(new URL('../../data/population.json', import.meta.url)).then(r => r.json());
```
Comment (thepassle) [inline src/services/populationService.js:6]: It would be nice to have some error handling and return a `status: 'failed'` if the request doesnt work. That way we can check for it here: https://github.com/Code-for-All/lockdown/blob/7f8e86f7e6f383c1ad9d75dea8449d692db4b937/src/components/CountryInfo.js#L96

**Comment by thepassle** [comment]
Currently this json is copied manually in a directory, right? Is this a file that will regularly change? Do we want to update it? if so: how do we go about that? Do we need an extra node script for it?

Also, the json file is unminified, depending on if we will update this file frequently, we will need to find a strategy to minify it. (If we dont frequently update it, we will have to minify it only once)

**Comment by thepassle** [comment]
Could you run `npm run format` before pushing?

File: data/population.json (hunk 1 above)
Comment (isunix11) [reply data/population.json:1051]: yup, will update

File: src/services/populationService.js (hunk 3 above)
Comment (isunix11) [reply src/services/populationService.js:6]: sure, will do

**Comment by isunix11** [comment]
we don't update because it's 2018 population census

**Comment by isunix11** [comment]
we will have to minify it once

File: src/components/CountryInfo.js (hunk 2 above)
Comment (isunix11) [reply src/components/CountryInfo.js:129]: I tested and it didn't display the offline message

Comment (thepassle) [reply src/components/CountryInfo.js:129]: it should display the offline message instead when a user is offline

Comment (isunix11) [reply src/components/CountryInfo.js:129]: yup, will fix that

File: src/components/CountryInfo.js [hunk 4]
```diff
@@ -92,7 +92,14 @@ export class CountryInfo extends Component {
...
     /** If the user is offline, and theres no response, or the response has failed */
-    if (!navigator.onLine) {
+
+    if (!lockdowns && !travelAdvice && !coronaData && navigator.onLine) {
+      return html`
+        Loading...
+      `;
+    }
+
+    if (navigator.onLine) {
```
Comment (thepassle) [inline src/components/CountryInfo.js:102]: Could you explain what the bug was and why these changes are made? Im a bit confused

Comment (thepassle) [reply src/components/CountryInfo.js:102]: I just checked the branch out locally, when you go offline and click a country, there is an error:

image

Comment (thepassle) [reply src/components/CountryInfo.js]: See https://github.com/Code-for-All/lockdown/pull/115/commits/0d71bedfa66c7b587ebdf73dd9b949a66684456c

This should show the appropriate success, error, and offline states

Comment (isunix11) [reply src/components/CountryInfo.js]: The Api's were offline when I was fixing. They are online now. I will update the code

Comment (isunix11) [reply src/components/CountryInfo.js]: The bug was in the populationService.js.  the return statement wasn't in the right place.
